docs/src/resolution_independence.md

```diff
@@ -0,0 +1,74 @@
+# Resolution independence
+
+## What is a pixel anyway?
+
+Pixel is short for *picture element* and although due to its popularity
+it has many meanings depending on context, when talking about pixels in the context of druid
+a pixel means always only one thing. It is **the smallest configurable area of color
+that the underlying platform allows `druid-shell` to manipulate**.
+
+The actual physical display might have a different resolution from what the platform knows or uses.
+Even if the display pixel resolution matches the platform resolution,
+the display itself can control even smaller elements than pixels - the sub-pixels.
+
+The shape of the physical pixel could be complex and definitely varies from display model to model.
+However for simplicity you can think of a pixel as a square which you can choose a color for.
+
+## Display pixel density
+
+As technology advances the physical size of pixels is getting smaller and smaller.
+This allows display manufacturers to put more and more pixels into the same sized screen.
+The **pixel densities of displays are increasing**.
+
+There is also an **increasing variety in the pixel density** of the displays used by people.
+Some might have a brand new *30" 8K UHD* (*7680px \* 4320px*) display,
+while others might still be rocking their *30" HD ready* (*1366px \* 768px*) display.
+It might even be the same person on the same computer with a multi-display setup.
+
+## The naive old school approach to UI
+
+For a very long time UIs have been designed without thinking about pixel density at all.
+People tended to have displays with roughly similar pixel densities, so it all kind of
+worked most of the time. However **it breaks down horribly** in a modern world.
+The *200px \* 200px* UI that looks decent on that *HD ready* display is barely visible
+on the *8K UHD* display. If you redesign it according to the *8K UHD* display then
+it won't even fit on the *HD ready* screen.
+
+## Platform specific band-aids
+
+Some platforms have mitigations in place where that small *200px \* 200px* UI
+will get scaled up by essentially **taking a screenshot of it and enlarging the image.**
+This will result in a blurry UI with diagonal and curved lines suffering the most.
+There is more hope with fonts where the vector information is still available to the platform,
+and instead of scaling up the image the text can be immediately drawn at the larger size.
```

Comment:
Not sure *blurry* is the right term. At least on a 200% display, Windows uses nearest-neighbour scaling rather than bilinear or such, so it’s just like you had half the resolution, rather than being *blurry* as such.

I’d tend to say it’s actually text that suffers the most. On Windows by default, either the app is DPI-aware, or the entire thing gets scaled. You can also override the app’s preference and opt into one of these modes (“application” and “system”) if you know how (right click on the executable or shortcut in Explorer, Properties, Compatibility, Change high DPI settings), and there’s a third mode you can choose these days, “system enhanced”, which tries to be clever, matching what you describe, so that sizes mostly get doubled, bitmaps get nearest-neighbour scaled, and GDI text gets rendered perfectly; it works fairly well on *some* apps (I run BPBible that way: it has minor issues and a few widgets end up half sized, but I’m OK with that), but badly on others (it made Audacity almost entirely half sized and too hard to use; and my experience is that DPI-unaware GTK apps are generally lousy in this mode).

Reply:
[Microsoft](https://support.microsoft.com/en-us/help/3025083/windows-scaling-issues-for-high-dpi-devices) seems to use *blurry* and *fuzzy* to describe the situation. I think it's fine, but if you have specific suggestions for replacement terms, I'm open to considering them.